### PR TITLE
(PC-7677) App native - Favoris - Afficher un message d'erreur à l'ajo…

### PR DESCRIPTION
**Pull request by kopax-polyconseil**
…ut des favoris si trop de favoris

Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-7677
## Checklist

I have:

- [x] Made sure the title of my PR follows the convention `[PC-XXX] <summary>`.
- [x] Made sure my feature is working on the relevant real / virtual devices.
- [x] Written **unit tests** for my feature.
- [ ] Added new reusable components to the AppComponents page and communicate to the team on slack
- [ ] Added a **screenshot** for UI tickets.
- [ ] Attached a **ticket number** for any added TODO/FIXME \
       (for tech tasks, give the best context about the TODO resolution: what/who/when).

## Screenshots

| \*iOS - [Phone name] | \*Android - [Phone name] | Tablet - [Phone name] | Little - [Phone name] |
| -------------------- | :----------------------: | --------------------: | --------------------: |
|                      |                          |                       |                       |

## Deploy hard

If native code (ios/android) was modified, **after** the PR is merged, on the master branch, upgrade the **app version** (+1 patch):

- if you want an hard deployment of the testing environment, use `yarn trigger:testing:deploy`

### CODE REVIEW

File: src/features/offer/components/OfferHeader.tsx [hunk 1]
```diff
@@ -43,9 +44,14 @@ export const OfferHeader: React.FC<Props> = (props) => {
...
       analytics.logHasAddedOfferToFavorites(params.from)
     },
-    onError: () => {
+    onError: (error) => {
       showErrorSnackBar({
-        message: _(t`L'offre n'a pas été ajoutée à tes favoris`),
+        message:
+          isApiError(error) && error.content.code === 'MAX_FAVORITES_REACHED'
+            ? _(
+                t`Trop de favoris enregistrés. Supprime des favoris pour pouvoir en ajouter de nouveaux`
```
Comment (BenStirrup) [inline src/features/offer/components/OfferHeader.tsx:52]: Petite faute sur "de nouveau" (j'admets que j'ai dû vérifier moi-même avant, donc pas facile à spotter :D) 
```suggestion
                t`Trop de favoris enregistrés. Supprime des favoris pour en ajouter de nouveau`
```

**Review by BenStirrup — APPROVED** [review]
Wording à modifier, sinon ok pour moi

File: src/features/offer/components/OfferHeader.tsx (hunk 1 above)
Comment (kopax-polyconseil) [reply src/features/offer/components/OfferHeader.tsx:52]: C'est un copier coller, est-ce qu'on fait valider avant quand même ?

Comment (BenStirrup) [reply src/features/offer/components/OfferHeader.tsx]: I see. Je te laisse décider

Comment (kopax-polyconseil) [reply src/features/offer/components/OfferHeader.tsx]: Lina est OK

**Pull request merged**